packages/p2p/src/components/orders/order-details/order-details-footer.jsx

```diff
@@ -1,198 +1,346 @@
-                    values={{ support_email: `p2p-support@${email_domain}`, id }}
-                    components={[
-                        <a
-                            key={0}
-                            className='link order-details__popup--danger'
-                            rel='noopener noreferrer'
-                            target='_blank'
-                            href={`mailto:p2p-support@${email_domain}`}
-                        />,
-                        <span key={1} className='order-details__popup--bold' />,
-                    ]}
-                />
-            ),
-            confirm_text: localize('Close'),
-            onClickConfirm: () => cancelPopup(),
-        };
+    return (
+        <Modal
+            className='confirm-modal'
+            is_open={should_show_confirm_modal}
+            toggleModal={hideConfirmOrderModal}
+            title={is_buy_order_for_user ? localize('Confirm payment?') : localize('Have you received payment?')}
+            has_close_icon
+        >
+            <Modal.Body>
+                {is_buy_order_for_user
+                    ? localize("Please make sure that you've paid {{amount}} {{currency}} to {{other_user_name}}.", {
+                          amount: price_display,
+                          currency: local_currency,
+                          other_user_name: other_user_details.name,
+                      })
+                    : localize(
+                          'Please confirm only after checking your bank or e-wallet account to make sure you have received payment.'
+                      )}
 
-        showPopup(options);
-    };
+                <Checkbox
+                    className='order-details__modal-checkbox'
+                    onChange={() => setIsCheckboxChecked(!is_checkbox_checked)}
+                    defaultChecked={is_checkbox_checked}
+                    label={
+                        is_buy_order_for_user
+                            ? localize('I have paid {{amount}} {{currency}}.', {
+                                  amount: price_display,
+                                  currency: local_currency,
+                              })
+                            : localize('I have received {{amount}} {{currency}}.', {
+                                  amount: price_display,
+                                  currency: local_currency,
+                              })
+                    }
+                />
+            </Modal.Body>
+            <Modal.Footer>
+                {error_message && <FormError message={error_message} />}
+                <Button.Group>
+                    <Button secondary type='button' onClick={hideConfirmOrderModal} large>
+                        {is_buy_order_for_user ? localize("I haven't paid yet") : localize('Cancel')}
```

Comment:
Try to use the `<Localize>` component over functions where possible.

Reply:
Done, thank you! 😄